whylog/assistant/span_list.py

```diff
@@ -0,0 +1,80 @@
+from operator import attrgetter
+from whylog.assistant.span import Span
+
+
+class OverlappingSpansError(Exception):
+    def __init__(self, span1, span2):
+        self.span1 = span1
+        self.span2 = span2
+
+    def __str__(self):
+        return "%s overlaps %s" % (self.span1, self.span2)
+
+
+class SpanList(list):
+    def __add__(self, other):
+        return SpanList(super(SpanList, self).__add__(other))
+
+    def sort_by_start(self):
+        return SpanList(sorted(self, key=attrgetter('start')))
+
+    def sort_as_date(self):
```

Comment:
It is not sorting by date. It is sorting by length, and reversed. by date could be confusing.